packages/jest-mock/src/index.ts

```diff
@@ -1003,27 +995,27 @@ export class ModuleMocker {
     return fn;
   }
 
-  spyOn<T extends {}, M extends NonFunctionPropertyNames<T>>(
```

Comment:
Intuitively it feels like `{}` type is somewhat similar to `{}` value in JS, but as a type it simply allows any object with any values. In other words, this is similar to `any` type. As a constrain it does nothing, safe to remove here. See https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492

Reply:
Hm.. `extends object` might be good here. Hard to say without type tests ;D Perhaps it is good idea to revert for now and to come back later with tests. Working on these.